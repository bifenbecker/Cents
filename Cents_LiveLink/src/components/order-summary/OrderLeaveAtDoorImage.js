import React from "react";
import {Flex, Image} from "rebass/styled-components";
import {DockModal} from "../common";
import useWindowSize from "../../hooks/useWindowSize";
import {useMemo} from "react";
import theme from "../../theme";

const OrderLeaveAtDoorImage = props => {
  const {isOpen, toggle, imageUrl} = props;
  const [height] = useWindowSize();

  const getSize = useMemo(() => {
    if (height >= 568) {
      return 250;
    } else {
      return 0.7 * height;
    }
  }, [height]);

  const handleBackClick = () => {
    toggle();
  };

  return (
    <DockModal
      header="Your order has been left at your door"
      isOpen={isOpen}
      toggle={toggle}
      size={getSize}
      fixedSize
      provideBackOption
      headerTextColor={theme.colors.primary}
      onBackClick={handleBackClick}
    >
      <Flex {...styles.deliveredImageContainer}>
        <Image
          src={imageUrl}
          style={styles.imageStyle}
          alt={"Image At The Door"}
          onClick={() => window.open(imageUrl, "_blank")}
        />
      </Flex>
    </DockModal>
  );
};

const styles = {
  mainWrapper: {
    sx: {
      height: "100%",
      flexDirection: "column",
    },
  },
  deliveredImageContainer: {
    sx: {
      alignItems: "center",
      justifyContent: "center",
    },
    width: "100%",
  },
  imageStyle: {
    width: "120px",
  },
};
export default OrderLeaveAtDoorImage;
