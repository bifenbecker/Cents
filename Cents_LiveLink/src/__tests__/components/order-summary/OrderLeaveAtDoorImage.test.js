import React from "react";
import {fireEvent, render} from "@testing-library/react";

import OrderLeaveAtDoorImage from "../../../components/order-summary/OrderLeaveAtDoorImage";

const imageUrl = "https://cdn.filestackcontent.com/u1tDDno0S1qbHRVOIm47";
const toggle = jest.fn();

const renderOrderLeaveAtDoorImagePopup = () => {
  const {getByText, container, getByAltText} = render(
    <OrderLeaveAtDoorImage
      {...{
        isOpen: true,
        toggle,
        imageUrl,
      }}
    />
  );

  return {getByText, getByAltText, container};
};

describe("DeliveryScheduledPopup Component", () => {
  it("should match snaphot", () => {
    const {container} = renderOrderLeaveAtDoorImagePopup();

    expect(container).toMatchSnapshot();
  });

  it("should render header", () => {
    const {getByText} = renderOrderLeaveAtDoorImagePopup();

    expect(getByText("Your order has been left at your door")).toBeInTheDocument();
  });

  it("should render back icon in the header bar", () => {
    const {getByAltText} = renderOrderLeaveAtDoorImagePopup();

    const imageAltText = "Dock Close";
    const icon = getByAltText(imageAltText);

    expect(icon).toBeInTheDocument();
  });

  it("should trigger toggle function when back icon is clicked", () => {
    const {getByAltText} = renderOrderLeaveAtDoorImagePopup();

    const imageAltText = "Dock Close";
    const icon = getByAltText(imageAltText);

    fireEvent.click(icon);
    expect(toggle).toBeCalled();
  });

  it("should render image taken by the driver while delivering the order", () => {
    const {getByAltText} = renderOrderLeaveAtDoorImagePopup();

    const imageAltText = "Image At The Door";
    const image = getByAltText(imageAltText);

    expect(image).toBeInTheDocument();
  });

  it("should open the image taken by driver driver in a new tab when imagee is clicked", () => {
    const {getByAltText} = renderOrderLeaveAtDoorImagePopup();
    jest.spyOn(window, "open");

    const imageAltText = "Image At The Door";
    const image = getByAltText(imageAltText);

    fireEvent.click(image);
    expect(window.open).toHaveBeenCalledWith(imageUrl, "_blank");
  });
});
