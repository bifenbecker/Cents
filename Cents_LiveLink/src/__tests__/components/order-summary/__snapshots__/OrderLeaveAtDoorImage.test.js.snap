// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeliveryScheduledPopup Component should match snaphot 1`] = `
<div>
  <div
    style="position: fixed; width: 0px; height: 0px; top: 0px; left: 0px; z-index: 99999999;"
  >
    <div
      style="position: fixed; left: 0px; right: 0px; top: 0px; bottom: 0px; z-index: 0; opacity: 1; transition: opacity 0.2s ease-out; background: white;"
    />
    <div
      style="position: fixed; z-index: 1; box-shadow: 0 0 4px rgba(0, 0, 0, 0.3); background: white; left: 25%; top: 0px; width: 50%; height: 768px; transition: left 0.2s ease-out,top 0.2s ease-out,width 0.2s ease-out,height 0.2s ease-out;"
    >
      <div
        style="position: absolute; z-index: 2; opacity: 0; top: -5px; height: 10px; left: 0px; width: 100%; cursor: row-resize;"
      />
      <div
        style="width: 100%; height: 100%; overflow: auto;"
      >
        <div
          class="sc-bczRLJ sc-gsnTZi haujOk hzPtvw"
          font-size="18px"
          height="67px"
        >
          <img
            alt="Dock Close"
            class="sc-bczRLJ FYQvo"
            src="Icon_Back.svg"
          />
          <div
            class="sc-bczRLJ fmojba"
            color="#3D98FF"
            font-size="18px"
          >
            Your order has been left at your door
          </div>
        </div>
        <div
          class="sc-bczRLJ sc-gsnTZi ielmAb hzPtvw"
          width="100%"
        >
          <img
            alt="Image At The Door"
            class="sc-bczRLJ iQjTFr"
            src="https://cdn.filestackcontent.com/u1tDDno0S1qbHRVOIm47"
            style="width: 120px;"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
